### DISCUSSION TITLE
2.4 ELRS range capabilities with telemetry ON

### DISCUSSION THREAD

**Discussion by Martivip** [ExpressLRS/ExpressLRS#732] [Q&A]
Hi All
I'm wondering about range capabilities ELRS 2.4 firmware with telemetry tuned ON, there is a lot of videos where 2.4 ELRS in 250Hz mode breaks out 33 km range but with telemetry OFF. After enabling the telemetry ON there will be a decent degradation of the range capabilities ? in general in 2.4 250 Hz mode there is enough bandwidth for telemetry ?

**Comment by JBKingdon** [comment]
Hi @Martivip 
Telemetry won't affect the range of the control link. The actual telemetry will likely drop out before the control link does since most receivers don't have as much output power as the transmitters, but whether you are sending telemetry or not shouldn't change the range of the outbound control link.

**Reply by Martivip** [reply]
Hi @JBKingdon i did a some research after posting this question and right now I'm little bit wiser :), right now my understanding is that enabling telemetry we are sacrificing control packets for telemetry so that's means that latency is temporary increased when telemetry packets are received by TX because TX need to send one bit confirmation of each received telemetry packet right ? 
And of course ELRS is flexible with possibility of choosing the telemetry ratio (More telemetry data impacts the latency because more need to be confirmed by TX and that confirmation sacrifices the control packets) but I'm wondering how each telemetry ratio on all Air rates increases the latency, there is a way to calculate this ? 
In example on 250 Hz mode with telemetry ratio 1:16 - how the latency will be increased due to telemetry ratio 1:16 ?

**Reply by JBKingdon** [reply]
Yes, a telemetry packet takes the place of a control packet. Basic telemetry doesn't use any additional confirmation bits, and for advanced telemetry they are squeezed into the normal RC packets, so confirmation bits don't add to latency. But yes, when a telemetry packet is sent instead of an RC packet then control latency is increased for that step, in the same way as if a packet was dropped due to interference. You can log rc_smoothing_rate to see the effect of dropped packets on packet interval.

**Reply by Martivip** [reply]
ok i see the light in the tunnel to understand this, so when a telemetry packet is sent from RX (in drone) to TX then TX need to stop transmitting the control packets to RX (in drone) due to reception of the telemetry packets and that's why latency is increased in that step, right ?

**Reply by JBKingdon** [reply]
Yes, that's right. There's enough time to send either a control packet tx -> rx, or a telemetry packet rx -> tx, but not both.

**Reply by Martivip** [reply]
I've notice that current packet ms rate is doubled when TX receives telemetry data of course with the interval related to set telemetry rate, but my question is that packet ms rate is always doubled ? not more or less ?

**Reply by JBKingdon** [reply]
That's right, at least under 'normal' conditions. If packets are also being dropped due to interference you might see larger intervals. The tx sends packets at fixed intervals based on a timer interrupt, so if a RC packet is replaced with a telemetry packet the interval to the next RC packet is exactly 2x the normal interval.

**Reply by Martivip** [reply]
now clear, you've helped me a lot Thank You
